lib/utils/job-utils/command-util.js

```diff
@@ -96,11 +96,27 @@ function commandUtilFactory(
                     result.source + ', ' + result.error,
                     { error: result });
             } else if (result.store) {
-                return waterline.catalogs.create({
+                var dataToSave = {
                     node: self.nodeId,
                     source: result.source || 'unknown',
                     data: result.data
-                });
+                };
+                if (self.updateExistingCatalog) {
+                    var query = {
+                        node: self.nodeId,
+                        source: result.source
+                    };
+                    return waterline.catalogs.count(query)
+                        .then(function(count) {
+                            if (count) {
+                                return waterline.catalogs.update(query, dataToSave);
```

Comment:
For our current design, there may be multiple catalog data from the same source because we don't remove the out-of-date data. For your choosed `update` command, all the out-of-date data (with same source) with be updated to  the exact same, if so, seems you should remove all duplicated ones, because it's useless to keep those identical data.